cpp/include/cudf/io/writers.hpp

```diff
@@ -178,9 +178,15 @@ class writer {
    *
    * @param table Set of columns to output
    * @param metadata Table metadata and column names
+   * @param return_filemetadata If true, return the raw file metadata
+   * @param metadata_out_file_path Column chunks file path to be set in the raw output metadata
    * @param stream Optional stream to use for device memory alloc and kernels
    */
-  void write_all(table_view const& table, const table_metadata *metadata = nullptr, cudaStream_t stream = 0);
+  std::unique_ptr<std::vector<uint8_t>> write_all(table_view const& table,
+                 const table_metadata *metadata = nullptr,
+                 bool return_filemetadata = false,
+                 const std::string metadata_out_file_path = "",
```

Comment:
I imagine the final destination for the metadata output file could end up being a cloud storage system like s3 or hdfs where maybe it would make more sense for this to use a similar abstraction as the IO sinks?

Reply:
It's really just a ~200-byte blob that dask will aggregate accross worker, so it's most likely going to be sent over the network (most use-cases will not need/use it). Not sure we'd want to force the client to implement a 2nd custom sink for this.

Reply:
I'm talking about specifically the `metadata_out_file_path` parameter. I guess I'm really unclear what this parameter is.

Reply:
Oh, `metadata_out_file_path` is actually the path being written in the parquet column chunk metadata: in normal cases, this path is not set, but dask will set this path to be the path of the parquet file being written, so when partitioning across multiple files, the collected metadata indicates which rowgroup is in which file (for example "partition1/data1.parquet").

Reply:
That makes a whole lot more sense. Resolving.